src/test/scala/org/broadinstitute/dsde/workbench/sam/api/UserRoutesV1Spec.scala

```diff
@@ -38,6 +39,36 @@ class UserRoutesV1Spec extends UserRoutesSpecHelper{
     }
   }
 
+  it should "create a user if ToS is enabled and the user specifies the ToS body correctly" in withTosEnabledRoutes { samRoutes =>
+    val tos = TermsOfServiceAcceptance("app.terra.bio/#terms-of-service")
+    Post("/register/user/v1", tos) ~> samRoutes.route ~> check {
+      status shouldEqual StatusCodes.Created
+      val res = responseAs[UserStatus]
+      res.userInfo.userSubjectId.value.length shouldBe 21
+      res.userInfo.userEmail shouldBe defaultUserEmail
+      res.enabled shouldBe Map("ldap" -> true, "allUsersGroup" -> true, "google" -> true)
+    }
+
+    Post("/register/user/v1", tos) ~> samRoutes.route ~> check {
+      status shouldEqual StatusCodes.Conflict
+    }
+  }
+
+  it should "forbid the registration if ToS is enabled and the user doesn't specify a ToS body" in withTosEnabledRoutes { samRoutes =>
+    Post("/register/user/v1") ~> samRoutes.route ~> check {
+      status shouldEqual StatusCodes.Forbidden
+      responseAs[ErrorReport].message should startWith("You must accept the Terms of Service in order to register.")
+    }
+  }
+
+  it should "forbid the registration if ToS is enabled and the user doesn't specify the correct ToS url" in withTosEnabledRoutes { samRoutes =>
+    val tos = TermsOfServiceAcceptance("onemillionpats.com")
+    Post("/register/user/v1", tos) ~> samRoutes.route ~> check {
+      status shouldEqual StatusCodes.Forbidden
+      responseAs[ErrorReport].message should startWith("You must accept the Terms of Service in order to register.")
+    }
+  }
```

Comment:
I suggest another test that posts a payload that won't unmarshal to a `TermsOfServiceAcceptance` class - for instance if the user posted some invalid json, or if they posted some random non-TermsOfServiceAcceptance case class